packages/transaction-controller/src/helpers/MultichainTrackingHelper.test.ts

```diff
@@ -222,6 +222,16 @@ function newMultichainTrackingHelper(
 describe('MultichainTrackingHelper', () => {
   beforeEach(() => {
     jest.resetAllMocks();
+
+    for (const network of [
+      'mainnet',
+      'goerli',
+      'sepolia',
+      'customNetworkClientId-1',
+    ] as const) {
+      MOCK_BLOCK_TRACKERS[network] = buildMockBlockTracker(network);
```

Comment:
The problem this solves is a good example of why sharing variables among tests by initializing them outside of all tests can lead to misleading and difficult to debug behavior. This fix makes sense, but if we really need specific block trackers and providers it would be better to create them on the fly in tests, using a helper function to assist. See here for more: https://github.com/MetaMask/contributor-docs/blob/main/docs/unit-testing.md#reset-shared-variables

Reply:
Good morning